### ISSUE TITLE
Dynamic layouts

### ISSUE THREAD

**Issue by vinothkbest** [nuxt/nuxt#27415]
**Hi Nuxt team**, 
Kindly add features in the framework for generating dynamic layouts /layouts/[...layoutName].vue like /pages/[...slug].vue

**Comment by manniL** [comment]
Hey @vinothkbest 

Please add some more information to your feature request (we have a feature request issue template for exactly that reason ☺️) including *why* you want that feature / what problem it solves that can't be solved at the moment.

<img width="884" alt="image" src="https://github.com/nuxt/nuxt/assets/640208/592d3d22-61f3-41e5-a503-5fd0fda93b4b">

https://github.com/nuxt/nuxt/issues/new?assignees=&labels=pending+triage%2C3.x&projects=&template=feature-request.yml

**Comment by danielroe** [comment]
@vinothkbest Why not just avoid the use of layouts and instead handle the logic in your `app.vue`?

**Comment by vinothkbest** [comment]
Yes. Absolutely awesome. Some pages having one layout, some pages having another layout. How to send json details in App.vue while pages rendering by [...slug].vue file. 

How to resolve it?

**Comment by danielroe** [comment]
There is a `setPageLayout` helper that you can use to change the layout programmatically. (However, note that in SSR this must be called _before_ rendering the layout, so you should probably call this in a plugin.)

**Comment by vinothkbest** [comment]
Yes you told correct. I can set layout dynamically but we need dynamic runtime layout file. 

Now, pre defined layouts only in nuxt3 like custom.vue, default.vue, desktop/index.vue in layouts directory. I want its generic one
